src/storage/BaseProcessor.h

```diff
@@ -98,6 +98,18 @@ class BaseProcessor {
         }
     }
 
+    void handleLeaderChanged(GraphSpaceID spaceId, PartitionID partId) {
+        auto addrRet = kvstore_->partLeader(spaceId, partId);
+        if (ok(addrRet)) {
+            auto leader = value(std::move(addrRet));
+            this->pushResultCode(cpp2::ErrorCode::E_LEADER_CHANGED, partId, leader);
+        } else {
+            LOG(ERROR) << "Fail to get part leader, spaceId: " << spaceId
+                       << ", partId: " << partId << ", ResultCode: " << error(addrRet);
+            this->pushResultCode(cpp2::ErrorCode::E_LEADER_CHANGED, partId);
```

Comment:
Return the real error code (error(addrRet)) instead of E_LEADER_CHANGED here.

Reply:
> Return the real error code (error(addrRet)) instead of E_LEADER_CHANGED here.
Good catched